janus_core/helpers/utils.py

```diff
@@ -229,3 +231,89 @@ def dict_remove_hyphens(dictionary: dict) -> dict:
         if isinstance(value, dict):
             dictionary[key] = dict_remove_hyphens(value)
     return {k.replace("-", "_"): v for k, v in dictionary.items()}
+
+
+def results_to_info(
+    struct: Atoms,
+    *,
+    properties: Collection[Properties] = (),
+    invalidate_calc: bool = False,
+) -> None:
+    """
+    Copy or move MLIP calculated results to Atoms.info dict.
+
+    Parameters
+    ----------
+    struct : Atoms
+        Atoms object to copy or move calculated results to info dict.
+    properties : Collection[Properties]
+        Properties to copy from results to info dict. Default is ().
+    invalidate_calc : bool
+        Whether to remove all calculator results after copying properties to info dict.
+        Default is False.
+    """
+    if not properties:
+        properties = get_args(Properties)
+
+    if struct.calc:
+        # Set default architecture from calculator name
+        arch = struct.calc.parameters["arch"]
+        struct.info["arch"] = arch
+
+        for key in properties & struct.calc.results.keys():
+            tag = f"{arch}_{key}"
+            value = struct.calc.results[key]
+            if key == "forces":
```

Comment:
Are the ASE compatible labels already in there?

Reply:
At a point where we'd want to copy/move them to `Atoms.info`, they'd be in `Atoms.calc.results`, if that's what you're asking.

The dictionary is automatically populated by running calculations e.g.

```
In [26]: x.calc.results
Out[26]: {}

In [27]: x.get_potential_energy()
^[[AOut[27]: -27.035127799332745

In [28]: x.calc.results
Out[28]:
{'energy': -27.035127799332745,
 'free_energy': -27.035127799332745,
 'node_energy': array([-0.86285876, -0.32487255, -0.86285876, -0.32487255, -0.86285876,
        -0.32487255, -0.86285876, -0.32487255]),
 'forces': array([[ 2.08166817e-17,  0.00000000e+00,  1.19298552e-20],
        [ 2.08166817e-17,  6.93889390e-18,  2.08504865e-17],
        [ 1.73472348e-17, -8.67361738e-19, -7.80625564e-18],
        [-6.93889390e-18, -1.73472348e-18,  5.20417043e-18],
        [ 1.73472348e-17,  1.04083409e-17, -1.04083409e-17],
        [-6.93889390e-18,  0.00000000e+00,  6.93889390e-18],
        [-8.67361738e-19,  2.60208521e-18, -5.06858339e-18],
        [-1.73472348e-18,  6.93889390e-18, -1.38777878e-17]]),
 'stress': array([-4.78327600e-03, -4.78327600e-03, -4.78327600e-03,  1.08000967e-19,
        -2.74004242e-19, -2.04504710e-19])}
```

With `write_results=True` for `ase.io.write`, these are then read back in with a generic `SinglePointCalculator` that make the same results still accessible:

```
In [35]: write(images=x, write_results=True, filename="test.xyz")

In [36]: y = read("test.xyz")

In [37]: y.calc
Out[37]: <ase.calculators.singlepoint.SinglePointCalculator at 0x7f9468830810>

In [38]: y.calc.results
Out[38]:
{'forces': array([[ 0.,  0.,  0.],
        [ 0.,  0.,  0.],
        [ 0., -0., -0.],
        [-0., -0.,  0.],
        [ 0.,  0., -0.],
        [-0.,  0.,  0.],
        [-0.,  0., -0.],
        [-0.,  0., -0.]]),
 'energy': -27.035127799332745,
 'free_energy': -27.035127799332745,
 'stress': array([-4.78327600e-03, -4.78327600e-03, -4.78327600e-03,  1.08000967e-19,
        -2.74004242e-19, -2.04504710e-19])}
```

Reply:
Sounds good. Just wanted to check.